### PR TITLE
🚀  Release v1.6.7

### PR DESCRIPTION
**Pull request by vishalanarase**
/area release

### CODE REVIEW

File: CHANGELOG/v1.6.7.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+## :sparkles: New Features
+- Dependency: Bump to Go 1.22.4 (#10802)
+
+## :bug: Bug Fixes
+- ClusterClass/MachineSet/MachineDeployment: MD/MS topo reconciler: only add finalizer for owned MD/MS (#10782)
+
+## :seedling: Others
+- clusterctl: Bump cert-manager to v1.15.1 (#10813)
+- Dependency: Bump to Go 1.22.5 (#10830)
+- Devtools: Makefile: pin controller-gen binary to go 1.21 (#10815)
```
Comment (chandankumar4) [inline CHANGELOG/v1.6.7.md:23]:
```suggestion
- Devtools: Pin controller-gen binary to go 1.21 (#10815)
```

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

/hold

File: CHANGELOG/v1.6.7.md (hunk 1 above)
Comment (vishalanarase) [reply CHANGELOG/v1.6.7.md]: Addressed

**Review by chandankumar4 — APPROVED** [review]

**Pull request merged**